src/content/github/file/main.tsx

```diff
@@ -60,20 +61,22 @@ function init(): Promise<void> {
 }
 
 async function main(): Promise<void> {
-  if (!isFileUrl(document.URL)) {
-    print("file not detected at current URL");
-    return;
-  }
+  try {
+    if (!isFileUrl(document.URL)) {
+      print("file not detected at current URL");
+      return;
+    }
 
-  let metadata: FileMetadata;
-  metadata = await getMetadata(document.URL);
+    let metadata: FileMetadata;
+    metadata = await getMetadata(document.URL);
 
-  globals.coverageButton = createCoverageButton();
+    globals.coverageButton = createCoverageButton();
 
-  process(metadata).catch((e) => {
-    print("unexpected error", e);
-    updateButton("Coverage: ⚠");
-  });
+    process(metadata)
+  } catch (e) {
+    Sentry.captureException(e)
+    throw e
```

Comment:
Im a _little bit_ concerned that before we used to just print the exceptions and now we're actually throwing them. Hoping this doesn't lead to some unintended behavior?

Reply:
I hadn't considered this actually. I think because the try catch is right at the top level call, we shouldn't have any side weird side effects. I did throw a bunch of test errors while working on this and saw nothing unexpected as a result. All of that said, I could be convinced to just continue printing the errors at this top-level error boundary. What do you think @ajay-sentry?

Reply:
The lower level throws I added only replaced early returns, and no unexpected behaviour should result there.